src/master-playlist-controller.js

```diff
@@ -998,37 +1038,59 @@ export class MasterPlaylistController extends videojs.EventTarget {
   /**
    * Update the player duration
    */
-  updateDuration() {
-    let oldDuration = this.mediaSource.duration;
-    let newDuration = Hls.Playlist.duration(this.masterPlaylistLoader_.media());
-    let buffered = this.tech_.buffered();
-    let setDuration = () => {
-      // on firefox setting the duration may sometimes cause an exception
-      // even if the media source is open and source buffers are not
-      // updating, something about the media source being in an invalid state.
-      this.logger_(`Setting duration from ${this.mediaSource.duration} => ${newDuration}`);
-      try {
```

Comment:
The try catch here was moved into sourceUpdaters `actions.duration` handler. Most of the rest of this logic is reformatted but unchanged.